screens/PokemonList.tsx

```diff
@@ -1,29 +1,100 @@
 import React, { useEffect, useState } from 'react';
-import { StyleSheet, Text, View, Button } from 'react-native';
-
-const PokemonList = ({ navigation }) => {
-  return (
-    <View style={styles.container}>
-      <Text style={styles.header}>Pokemon List</Text>
-      {/* <Button
-        title='Go to Native Module'
-        onPress={() => navigation.navigate('NativeModule')}
-      /> */}
-    </View>
-  );
+import { StyleSheet, Text, View, FlatList, Dimensions } from 'react-native';
+import { getAllPokemons } from '../api/pokemon';
+
+const keyExtractor = ({ name }) => name;
+
+const PokemonList = () => {
+  const step = 20;
+  const [batchStart, setBatchStart] = useState(0);
+  const [loading, setLoading] = useState(true);
+  const [pokemons, setPokemons] = useState(null);
+
+  useEffect(() => {
+    const effect = async () => {
+      const allPokemons = await getAllPokemons(batchStart, step);
+      setPokemons(allPokemons);
+      setLoading(false);
+      setBatchStart(batchStart + step);
+    };
+    effect();
+  }, []);
+
+  const renderItem = ({ item }) => {
+    //const matchRes = item.url.match('(pokemon\\/{1}(\\d+)\\/{1})', 'gi');
+
+    // this is more readable than regex
+    const urlChopped = item.url.split('/');
+    const id = urlChopped[urlChopped.length - 2];
+    return (
+      <Text style={styles.header}>
+        {id}. {item.name}
+      </Text>
+    );
+  };
+
+  const scrolledCloseToEnd = ({
+    layoutMeasurement,
+    contentOffset,
+    contentSize
+  }) => {
+    return (
+      layoutMeasurement.height + contentOffset.y >= contentSize.height - 200
```

Comment:
why `200`? If it's a "magic number" then it's good to move it to `const` with appriopriate name :)

Reply:
yeah, it was 1 at first but I wanted the new content to load before we reach the bottom